### ISSUE TITLE
toImmutable method on Modifiable class not generated when copy = false

### ISSUE THREAD

**Issue by ocrease** [immutables/immutables#301]
I want to create a Modifiable companion object with the `toImmutable` method but I don't want my Immutable type to have the `with*` copy methods. Take this example:

```java
@Value.Immutable(copy = false)
@Value.Style(
    typeAbstract = "*Def",
    typeImmutable = "*",
    visibility = Value.Style.ImplementationVisibility.PUBLIC, 
    builderVisibility = Value.Style.BuilderVisibility.PUBLIC)
@Value.Modifiable
abstract class SampleDef
{
    public abstract int getA();
}
```

If I remove `copy = false` from the `Immutable` annotation, then the `toImmutable` method is generated on the modifiable class, but then I end up having all the `with*` methods that I don't want.

Is there a way to achieve that? I can't see why the two are related. It looks like the only requirement for the `toImmutable` method is that the `copyOf` method exists on the immutable class, and the `copyOf` method does not depend on the `with*` methods.

**Comment by elucash** [comment]
Currently `copy=false` also disables the copy constructor which in turn disables `toImmutable` method. I'll try to see how they are related and if it's possible to separate those concepts, to have `copyOf` and `toImmutable` generated even if `copy=false`. Thank you for reporting this!

**Comment by ocrease** [comment]
That's great, thanks for looking into it. I think the logic could be that if the object is annotated with `@Value.Modifiable`, then the copy constructor is generated regardless of whether `copy=false` or not. If I remember correctly the `copyOf` constructor is package private, so having it on the immutable class shouldn't be too concerning. Of course, I might have missed something! 

I've got to say I'm really impressed with the tool. It is very powerful and easy to customize! Thank you very much.

**Comment by elucash** [comment]
I decided to not change how configuration `copy=false` works, but now `toImmutable` will be generated regardless of `copyOf`.

**Comment by ocrease** [comment]
Hi @elucash, I pulled the latest version of the code and the generated code looks good. Thanks for the quick turnaround.

**Comment by elucash** [comment]
Released in v 2.1.16 Thank you!
